### PR TITLE
Discord Rich Presence Time Remaining and Elapsed Time + GUI menu changes

### PR DESCRIPTION
**Pull request by AsKMoose**
**Discord Rich Presence Features** 

Added time remaining based on in game round time.
Added time elapsed if game has no round time limit. 

remaining
elapsed

Big thanks to @iSupersniper, @KILLER-CHIEF, and @StelluR for helping me find the addresses I needed. 

**FOV and Crosshair Menu Changes** 

Replaced "FOV Alteration Disabled" with "Default FOV" in the custom menu since the previous menu didn't really disable FOV alteration but rather set it back to a default value. Did the same for crosshair offset. 

defaultFOV

**Misc. Changes** - corrected misspelled variable

### CODE REVIEW

**Comment by iSupersniper** [comment]
Please update this pull request, and provide a test dll in dev preview

**Comment by shayaantx** [comment]
Please update this pull request, and provide a test dll in dev preview

File: xlive/H2MOD/Modules/CustomMenu/CustomMenu.cpp [hunk 1]
```diff
@@ -1234,7 +1234,7 @@ const int CMLabelMenuId_EditCrosshair = 0xFF000010;
 #pragma region CM_EditCrosshair
 
 static void loadLabelCrosshairOffset() {
-	if (!FloatIsNaN(H2Config_crosshair_offset)) {
+	if (H2Config_crosshair_offset != 0.165f) {
```
Comment (shayaantx) [inline xlive/H2MOD/Modules/CustomMenu/CustomMenu.cpp:1238]: what does this have to do with your discord presence change? no scope creep

File: xlive/H2MOD.cpp [hunk 2]
```diff
@@ -39,6 +39,7 @@ bool b_XboxTick = false;
 bool b_Infection = false;
 bool b_Halo2Final = false;
 bool b_HeadHunter = false;
+bool b_inGame = false;
```
Comment (shayaantx) [inline xlive/H2MOD.cpp:39]: don't set booleans for whether you are in game or not, actually find the h2v engines way of determining if you are in-game

File: xlive/H2MOD.cpp [hunk 3]
```diff
@@ -1161,6 +1165,7 @@ void __cdecl OnMapLoad(int a1)
 		H2Tweaks::disable60FPSCutscenes(); 
 		H2Tweaks::setSavedSens();
 		H2Tweaks::sunFlareFix();
+
```
Comment (shayaantx) [inline xlive/H2MOD.cpp:1159]: please remove newline

File: xlive/H2MOD/Discord/DiscordInterface.cpp [hunk 4]
```diff
@@ -76,6 +84,35 @@ static void updateDiscordPresence()
 	discordPresence.largeImageText = map_mouse_over;
 	discordPresence.smallImageKey = mode_key;
 	discordPresence.smallImageText = mode_details;
+	if (b_inGame != false) //called only in MP engine
+	{
+		previousRoundNumber = *(int*)((BYTE*)0x300056C6);
```
Comment (shayaantx) [inline xlive/H2MOD/Discord/DiscordInterface.cpp:89]: please document what this random offset is (move the value into a constant value as well)

File: xlive/H2MOD/Discord/DiscordInterface.cpp [hunk 5]
```diff
@@ -76,6 +84,35 @@ static void updateDiscordPresence()
...
+		DWORD MPgameState = *(DWORD*)(H2BaseAddr + 0x514E48); //prevents startTimestamp and endTimestamp from being called in postgame
```
Comment (shayaantx) [inline xlive/H2MOD/Discord/DiscordInterface.cpp:90]: please be consistent with your variable camelcase

File: xlive/H2MOD/Discord/DiscordInterface.cpp [hunk 6]
```diff
@@ -76,6 +84,35 @@ static void updateDiscordPresence()
...
+		roundTimeExists = (DWORD)*(DWORD*)((DWORD)H2BaseAddr + 0x482d3c);
+		roundTimeCheck = *(DWORD*)(roundTimeExists + 0x8 + 0x2A0 + 0x54); //checks whether game has a round time
+
+		if (MPgameState != 1 && roundTimeCheck == 0)
```
Comment (shayaantx) [inline xlive/H2MOD/Discord/DiscordInterface.cpp:94]: actually doc what are all the mpgame states and what each one means

File: xlive/H2MOD/Discord/DiscordInterface.cpp (hunk 4 above)
Comment (shayaantx) [reply xlive/H2MOD/Discord/DiscordInterface.cpp:89]: preferably move them into functions that are readable

File: xlive/H2MOD/Discord/DiscordInterface.cpp [hunk 7]
```diff
@@ -136,6 +173,21 @@ void DiscordInterface::SetPlayerCountInfo(int current, int max)
...
+{
+	int currentRoundNumber = *(int*)((BYTE*)0x300056C6);
+	DWORD roundState = *(DWORD*)(0x3000629C); //takes on different values when transitioning from round end to new round
+	
+	if (currentRoundNumber != previousRoundNumber && roundTimeCheck != 0 && updateOnNewRound != true) {
+		updateDiscordPresence();
+		updateOnNewRound = true;
+	}
+
+	if (roundState == 1 && updateOnNewRound != false) {
```
Comment (shayaantx) [inline xlive/H2MOD/Discord/DiscordInterface.cpp:186]: what does roundState == 1 actually mean? actually document all the possible values for round state.

File: xlive/H2MOD/Discord/DiscordInterface.cpp (hunk 5 above)
Comment (shayaantx) [reply xlive/H2MOD/Discord/DiscordInterface.cpp:90]: no underscores

**Comment by shayaantx** [comment]
overall the way you figure out start/end time of a game seems like not the greatest way to accomplish what you are doing, you should:

1) hook into something during start of the game, set discord presence start time
2) hook into something during end of the game, set discord presence end time

if you want to open a new pr thats fine or do it under this one.
